lib/counter_culture/counter.rb

```diff
@@ -17,6 +17,8 @@ def initialize(model, relation, options)
       @delta_magnitude = options[:delta_magnitude] || 1
       @with_papertrail = options.fetch(:with_papertrail, false)
       @execute_after_commit = options.fetch(:execute_after_commit, false)
+      @if_conditions = Array.wrap(options[:if])
+      @unless_conditions = Array.wrap(options[:unless])
```

Comment:
Here's a different idea, I'd love to discuss whether that's a better approach: What if we took what's passed into `options[:if]` or `options[:unless]` and "compile" that into a `Proc` to pass to use for `column_name`? This is pseudo-code, but something like
```
if options[:if].present?
  @counter_cache_name = proc  do |model|
    Array.wrap(options[:if]).inject(:&&) { |method| model.public_send(method) }
  end
end
```

The advantage I see of that is that it simplifies the code paths—there's still only one code path to consider and `if` and `unless` truly just become shorthand.

Is that better? Or does that make the code more complicated?

Reply:
Yeah that was the approach I considered originally, but then we can't specify a custom column name when using `if`/ `unless`.

Reply:
Can you say a bit more about that? I don't think I follow why that doesn't work.

Reply:
Sure, if I’m understanding you correctly that we “compile” into a proc passed internally to column_name, then we can’t do the following:

column_name: :foo_count, if: :special?

Because our compiled proc for if would overwrite the :foo_count symbol

Reply:
Couldn't we use the value passed as the name in the proc we compile the `if` / `unless` into?

Reply:
Yeah, you’re right. Nice, I’ll take a look at refactoring to that.

Reply:
Updated, LMK what you think. Then I'll look at updating the tests